src/app/features/courses/courses-list/courses-list.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, EventEmitter, Input, Output } from "@angular/core";
+import {
+  IconDefinition,
+  faTrashCan,
+  faPencil,
+} from "@fortawesome/free-solid-svg-icons";
+interface Course {
```

Comment:
You are right to use this as a blueprint for course entities. However, this needs to be moved to somewhere else, maybe under a new 'models' folder under shared folder and accommodate all entity model related files there. Export them and use it via an import where ever needed.